### DISCUSSION TITLE
estimate the gene activity of a subset of peaks

### DISCUSSION THREAD

**Discussion by pengxin2019** [stuart-lab/signac#868] [Q&A] [answered]
Hi Tim,
In terms of the `GeneActivity` in this pipeline: https://satijalab.org/signac/articles/pbmc_vignette.html
I would like to estimate the gene activity of a subset of the peaks and create a RNA assay of these estimates. Then make a `featureplot` of these estimated gene activity to see how it is differentially acessbille across entire dataset.

```

An object of class Seurat 
153602 features across 17230 samples within 2 assays 
Active assay: ATAC (28471 features, 0 variable features)
 1 other assays present: peaks
 3 dimensional reductions calculated: integrated_lsi, umap, lsi

counts.integrated <- FeatureMatrix(
  fragments = fragments,
  features = granges(a genomic region object containing a subset peaks),  
  cells = Cells(integrated))

integrated[["ATAC"]] <- CreateChromatinAssay(
  counts = counts.integrated,
  fragments = Fragments(integrated),
  annotation = Annotation(integrated)
)
DefaultAssay(integrated) <- "ATAC"

gene.activities <- GeneActivity(integrated, extend.upstream = 2000) 
integrated[['RNA']] <- CreateAssayObject(counts = gene.activities)
DefaultAssay(integrated) <- 'RNA'

FeaturePlot(
  object = integrated,
  features = "CD14",
  pt.size = 0.03,
  label = FALSE,
  label.size = 1, ncol = 2, combine = TRUE, coord.fixed = FALSE)
#I only want to see the estimated gene expression of CD14 only based on the  subset of peaks instead of all peaks. 
```

But it seems like the gene.activities object is the same as when I use all peaks rather than subset of peaks in `granges` prarameter in  `FeatureMatrix` function. Is this because that I am still using all `fragments` when I made the `counts.integrated` by running FeatureMatrix? Can you give me some suggestion on how to do it or  if it is doable?

Thanks
Best
Penny

**Comment by timoast** [comment, marked as answer]
The `GeneActivity()` function doesn't use peaks, but quantifies the number of fragments in regions of the genome. See the documentation for `GeneActivity()` here: https://satijalab.org/signac/reference/geneactivity

We don't currently have a function to sum the accessibility of different peaks per cell
